src/IdbBackend.js

```diff
@@ -1,15 +1,66 @@
-const idb = require("idb-keyval");
+const idb = require("@wmhilton/idb-keyval");
+
+const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))
+
+const iam = typeof WorkerGlobalScope === "undefined" ? "main": "worker"
 
 module.exports = class IdbBackend {
   constructor(name) {
     this._database = name;
     this._store = new idb.Store(this._database, this._database + "_files");
+    this._saving = null
+    this._fetching = null
+    this._timeout
   }
-  saveSuperblock(superblock) {
-    return idb.set("!root", superblock, this._store);
+  async storeSuperblock(superblock) {
+    this._cache = superblock
+    await this._fetching
+    clearTimeout(this._timeout)
+    this._timeout = setTimeout(async () => {
+      let done
+      this._saving = new Promise(resolve => { done = resolve })
+      this._cache = null
+      await idb.set("!root", superblock, this._store);
+      await idb.del("!locked", this._store);
+      this._saving = null
+      console.log(`${iam} released lock`)
+      done()
+    }, 500)
   }
-  loadSuperblock() {
-    return idb.get("!root", this._store);
+  async fetchSuperblock() {
+    await this._saving
+    if (this._fetching) return this._fetching
+    if (this._cache) return this._cache
+    let done
+    this._fetching = new Promise(resolve => { done = resolve })
+    let locked = true
+    let call = Math.random()
+    while(locked) {
+      await idb.update("!locked", value => {
+        if (value === undefined) {
+          // console.log(`${iam} ${call} grabs the lock`)
+        } else {
+          // auto-expire locks after 24 hours
+          if (value < (new Date().valueOf() - 24 * 60 * 60 * 1000)) {
```

Comment:
I should expose a helper function to blast the lock manually.

Reply:
done. `overrideLock` can be used say, after a page crashes or is refreshed while in the middle of file operations